### PR TITLE
feat(txpool): More optimizations & tests

### PR DESCRIPTION
**Pull request by itsdevbear**

### CODE REVIEW

File: cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go [hunk 1]
```diff
@@ -122,6 +123,7 @@ func (etp *EthTxPool) queued() map[common.Address]coretypes.Transactions {
 //
 // NOT THREAD SAFE.
 func (etp *EthTxPool) Nonce(addr common.Address) uint64 {
+	exit := false
 	pendingNonces := make(map[common.Address]uint64)
```
Comment (calbera) [inline cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go:127]: there is no need to store a map of addresses actually, we only care about the given `addr`

File: cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go [hunk 2]
```diff
@@ -146,6 +148,10 @@ func (etp *EthTxPool) Nonce(addr common.Address) uint64 {
 			pendingNonces[addr]++
 		case txNonce > pendingNonce+1:
 			// As soon as we see a non contiguous nonce we break.
+			exit = true
+		}
+
+		if exit {
```
Comment (calbera) [inline cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go:154]: you can actually use a "label" to do this 
image

File: cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go (hunk 1 above)
Comment (itsdevbear) [reply cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go:129]: nah Stats() returns everything

Comment (calbera) [reply cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go:129]: what? this is just for the nonce function which only cares about the given `addr`

Comment (itsdevbear) [reply cosmos/x/evm/plugins/txpool/mempool/mempool_reader.go:129]: oh lol ui glitched

**Comment by itsdevbear** [comment]
replaced by #1041

**Pull request closed without merging**